src/lib/azure/storage.ts

```
@@ -0,0 +1,195 @@
+  try {
+    logger.info(`Finding ${message}`);
+    const client = await getStorageClient();
+    const accounts: StorageAccountsListByResourceGroupResponse = await client.storageAccounts.listByResourceGroup(
+      resourceGroup
+    );
+    let exists = false;
+
+    logger.debug(
+      `${accounts.length} storage accounts found in ${resourceGroup}`
+    );
+    for (const account of accounts) {
+      logger.debug(`Found ${account.name} so far`);
+      if (account.name === accountName) {
+        exists = true;
+        break;
+      }
+    }
+
+    // Storage account exists so exit the method.
+    if (exists) {
+      logger.info(`Found ${message}`);
+      return;
+    }
+    logger.info(`${message} does not exist`);
+    // Storage account does not exist so create it.
+    await createStorageAccount(resourceGroup, accountName, location);
+  } catch (err) {
+    logger.error(err);
+  }
+};
+
+/**
+ * Creates Azure storate account `name` in resource group `resourceGroup` in 1ocation `location`
+ *
+ * @param resourceGroup Name of Azure reesource group
+ * @param accountName The Azure storage account name
+ * @param location The Azure storage account location
+ */
+export const createStorageAccount = async (
+  resourceGroup: string,
+  accountName: string,
+  location: string
+) => {
+  const message = `Azure storage account ${accountName} in resource group ${resourceGroup} in ${location} location.`;
+  try {
+    logger.verbose(`Create storage client object.`);
+    const client = await getStorageClient();
+    logger.verbose(
+      `Checking for storage account name ${accountName} availability`
+    );
+    const response: StorageAccountsCheckNameAvailabilityResponse = await client.storageAccounts.checkNameAvailability(
+      accountName
+    );
+
+    if (response.nameAvailable === false) {
+      logger.error(
+        `Storage account name ${accountName} is not available. Please choose a different name.`
+      );
+      return;
+    }
```

Maybe we want to throw an error too?

---

That makes sense, added it.